atomic_reactor/plugins/exit_sendmail.py

```diff
@@ -231,8 +273,8 @@ def run(self):
                                         '", "'.join(sorted(unknown_states)))
 
         rebuild = is_rebuild(self.workflow)
-        success = not self.workflow.build_result.is_failed()
-        canceled = self.workflow.autorebuild_canceled
+        success = not self.workflow.build_process_failed
+        canceled = self.workflow.autorebuild_canceled or self.workflow.build_canceled
```

Comment:
No, the current behaviour is incorrect: the config can ask for manual cancellation but not automatic cancellation, and the implementation doesn't do that.

Why not just have `auto_canceled = self.workflow.autorebuild_canceled` and `manual_canceled = self.workflow.build_canceled` and pass both through to `_should_send()`?